packages/node_modules/node-red/red.js

```diff
@@ -130,6 +155,33 @@ try {
     process.exit();
 }
 
+if (parsedArgs.define) {
+    var defs = parsedArgs.define;
+    defs.forEach(function (def) {
+        try {
+            var match = /^(([^=]+)=(.+)|@(.*))$/.exec(def);
+            if (match) {
+                if (!match[4]) {
+                    var val = JSON.parse(match[3]);
+                    setProperty(settings, match[2], val);
+                }
+                else {
+                    var obj = fs.readJsonSync(match[4]);
+                    Object.entries(obj).forEach(([key, val]) => {
+                        settings[key] = val;
```

Comment:
I think the code needs to merge settings loaded from a file, rather than replace the entire  top-level property.

For example, to enable `editorTheme.projects.enabled` a user may have other settings under `editorTheme` that they want to keep.

Reply:
Changed to merge settings object when reading from a file.

Reply:
I've been playing with this, and I keep finding cases where merging is the right behaviour (for example, setting `editorTheme.projects.enabled` without overwriting everything else under `editorTheme`), and cases where the user would want to replace the value without merging. For example, if they wanted to apply a different `adminAuth` scheme, they would want to fully replace what was there and not merge.

How do we handle both cases? Do we even try to handle both cases?

I suggest we use a slightly different format for the custom settings file:

```
{
   "editorTheme.projects.enabled": true,
   "adminAuth": {
      ...
   }
}
```

Rather than copy the full structure of the settings file, it is an object whose keys identify the property to overwrite with the corresponding value.

This would allow an individual settings to be modified, as well as entire objects. I think this approach is the only way to satisfy the requirements.

I will work on this change as part of merging this PR.

Reply:
I thought it would be better if the override settings could be a simple subset of settings.js. However, I think you have a valid point, so I agree with your proposal.